lib/sparkql/parser_compatibility.rb

```diff
@@ -254,5 +259,13 @@ def range?(value)
   def operator_supports_multiples?(operator)
     OPERATORS_SUPPORTING_MULTIPLES.include?(operator)
   end
+  
+  def coerce_datetime datetime
+    if datestr = datetime.match(/^(\d{4}-\d{2}-\d{2})/)
+      datestr[0]
+    else
+      datetime
+    end
+  end
```

Comment:
It's been a long long time since I got a gold star :-D

The nesting was gonna be ugly without this little guy.